sphinx/writers/latex.py

```diff
@@ -1513,30 +1524,33 @@ def visit_entry(self, node):
                        '\\vbox{\\hbox{\\strut}}\\end{varwidth}%\n') + context
             self.needs_linetrimming = 1
         if len(node.traverse(nodes.paragraph)) >= 2:
-            self.table.has_oldproblematic = True
+            for i in range(cell.col, cell.col + cell.width):
+                self.table.empirical_colwidths[i] = max(self.table.empirical_colwidths[i], 17)
         if isinstance(node.parent.parent, nodes.thead) or (cell.col in self.table.stubs):
             if len(node) == 1 and isinstance(node[0], nodes.paragraph) and node.astext() == '':
                 pass
             else:
                 self.body.append('\\sphinxstyletheadfamily ')
-        if self.needs_linetrimming:
-            self.pushbody([])
+        self.pushbody([])
         self.context.append(context)
 
     def depart_entry(self, node):
         # type: (nodes.Node) -> None
+        body = self.popbody()
         if self.needs_linetrimming:
             self.needs_linetrimming = 0
-            body = self.popbody()
-
             # Remove empty lines from top of merged cell
             while body and body[0] == "\n":
                 body.pop(0)
-            self.body.extend(body)
 
+        self.body.extend(body)
         self.body.append(self.context.pop())
 
         cell = self.table.cell()
+        bodywidthpercol = float(sum(len(i) for i in body)) / cell.width
```

Comment:
I am not happy with this at it counts characters of LaTeX mark-up such as a footnote or `\textbackslash{}` which artificially increase weight of corresponding column

Reply:
What I would like is simply measure the source length for the cell contents, before LaTeX escapes or added mark-up; for lack of enough knowledge of docutils nodes, I used this alternative, which is not satisfactory.